include/gsl/gsl_byte

```diff
@@ -37,10 +37,30 @@
 #pragma push_macro("noexcept")
 #define noexcept /*noexcept*/
 #endif           // _MSC_VER <= 1800
+
+#if _HAS_STD_BYTE 
+
```

Comment:
There's unfortunately no SD-6 feature test macro for `std::byte`, which libc++ and libstdc++ would otherwise define appropriately. You should complain to the author of the `std::byte` proposal ;)